test/Microsoft.OpenApi.Tests/Walkers/WalkerLocationTests.cs

```diff
@@ -0,0 +1,229 @@
+                                        { "application/json", new OpenApiMediaType {
+                                                Schema = new OpenApiSchema
+                                                {
+                                                    Type = "string"
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    }
+                }
+            });
+
+            var locator = new LocatorVisitor();
+            var walker = new OpenApiWalker(locator);
+            walker.Walk(doc);
+
+            locator.Locations.ShouldBeEquivalentTo(new List<string> {
+                "#/info",
+                "#/servers",
+                "#/components",
+                "#/externalDocs",
+                "#/tags",
+                "#/paths",
+                "#/paths/~1test",
+                "#/paths/~1test/get",
+                "#/paths/~1test/get/responses",
+                "#/paths/~1test/get/responses/200",
+                "#/paths/~1test/get/responses/200/content",
+                "#/paths/~1test/get/responses/200/content/application~1json",
+                "#/paths/~1test/get/responses/200/content/application~1json/schema",
+                "#/paths/~1test/get/responses/200/content/application~1json/schema/externalDocs",
+            });
+        }
+
+        [Fact]
+        public void WalkDOMWithCycles()
+        {
+            var loopySchema = new OpenApiSchema()
+            {
+                Type = "object",
+                Properties = new Dictionary<string,OpenApiSchema>()
+                {
+                    { "name", new OpenApiSchema() { Type = "string" }
+                    }
+                }
+            };
+
+            loopySchema.Properties.Add("parent", loopySchema);
+
+            var doc = new OpenApiDocument();
+            doc.Components = new OpenApiComponents()
+            {
+                Schemas = new Dictionary<string, OpenApiSchema>
+                {
+                    { "loopy", loopySchema } 
+                }
+            };
+
+            var locator = new LocatorVisitor();
+            var walker = new OpenApiWalker(locator);
+            walker.Walk(doc);
+
+            locator.Locations.ShouldBeEquivalentTo(new List<string> {
+                "#/info",
+                "#/servers",
+                "#/paths",
+                "#/components",
+                "#/components/loopy",
+                "#/components/loopy/properties/name",
+                "#/components/loopy/properties/name/externalDocs",
+                "#/components/loopy/externalDocs",
+                "#/externalDocs",
+                "#/tags"
+            });
+        }
+    }
+
+    public class LocatorVisitor : OpenApiVisitorBase
```

Comment:
This is actually a pretty nice feature. What do you think about moving it to the source, instead of leaving it in tests?

Reply:
This doesn't have to be done with this PR though. Maybe let's change this to private and we can decide?